crates/bevy_animation/src/tracks/n.rs

```diff
@@ -0,0 +1,231 @@
+    pub len: u16,
+    pub track: TrackFixed<V>,
+}
+
+impl<V: ValueN + Lerp + Clone> TrackN for TrackFixedN<V> {
+    type Output = V::Value;
+
+    fn sample(&self, cursor: u16, time: f32, assign: &mut dyn FnMut(u16, Self::Output)) -> u16 {
+        let (cursor, x) = self.track.sample_with_cursor(cursor, time);
+        let x = x.unpack();
+        for i in 0..self.len() {
+            (assign)(*self.lanes.get(i), *x.get(i));
+        }
+        cursor
+    }
+
+    fn duration(&self) -> f32 {
+        self.track.duration()
+    }
+
+    fn len(&self) -> usize {
+        (self.len as usize).min(V::size())
+    }
+
+    fn lanes(&self) -> &[u16] {
+        &self.lanes.as_slice()[0..self.len()]
+    }
+
+    fn size(&self) -> usize {
+        V::size()
+    }
+}
+
+// impl<V: ValueN + Lerp + Clone> TrackFixedN<V> {
+//     // TODO: Function to edit each lane
+//     // pub fn add_track
+
+//     pub fn add_track_resampled(&mut self, track: &dyn Track<Output = <V as ValueN>::Value>) {
+//         let i = self.len as usize;
+//         if i >= V::size() {
+//             panic!("track is full");
+//         }
+
+//         self.len += 1;
+
+//         let mut cursor = 0;
+//         let f = 1.0 / (self.track.frame_rate() as f32);
+//         let offset = self.track.offset() as f32 * f;
+//         let s = self.track.len();
+//         let keyframes = self.track.keyframes_mut();
+
+//         for frame in 0..s {
+//             let time = offset + f * frame as f32;
+//             let (k, v) = track.sample_with_cursor(cursor, time);
+//             keyframes[frame].set(v, i);
+//             cursor = k;
+//         }
+//     }
+
+//     // pub fn remove_track
+// }
+
+pub type TrackNBase<T> = Box<dyn TrackN<Output = T> + Send + Sync + 'static>;
+
+///////////////////////////////////////////////////////////////////////////////
+
+macro_rules! valuen {
+    ($t:ty, $i:ty, $o:ty, $s:tt) => {
+        impl ValueN for $t {
+            type Value = $o;
+
+            type Lanes = [u16; $s];
+
+            type Outputs = [Self::Value; $s];
+
+            #[inline(always)]
+            fn pack(values: Self::Outputs) -> Self {
+                unsafe { transmute::<_, [$i; $s]>(values).into() }
+            }
+
+            #[inline(always)]
+            fn unpack(self) -> Self::Outputs {
+                unsafe { transmute::<[$i; $s], _>(self.into()) }
+            }
+
+            fn size() -> usize {
+                $s
+            }
+        }
+    };
+}
+
+valuen!(f32x4, f32, f32, 4);
+valuen!(f32x8, f32, f32, 8);
+valuen!(Vec3x4, ultraviolet::Vec3, Vec3, 4);
+valuen!(Vec3x8, ultraviolet::Vec3, Vec3, 8);
+valuen!(Vec4x4, ultraviolet::Vec4, Vec4, 4);
+valuen!(Vec4x8, ultraviolet::Vec4, Vec4, 8);
+
+impl ValueN for Quatx4 {
+    type Value = Quat;
+
+    type Lanes = [u16; 4];
+
+    type Outputs = [Self::Value; 4];
+
+    #[inline(always)]
```

Comment:
Please use `#[inline]` instead. That allows LLVM to not inline in debug mode or when it is consideres to not have an advantage.